vnext/ReactUWP/UwpPreparedScriptStore.cpp

```diff
@@ -0,0 +1,106 @@
+};
+
+namespace react { namespace uwp {
+  UwpPreparedScriptStore::UwpPreparedScriptStore(winrt::hstring uri)
+  {
+    if (!uri.empty())
+    {
+      m_byteCodeFileAsync = winrt::StorageFile::GetFileFromApplicationUriAsync(winrt::Uri(uri));
+    }
+  }
+
+std::unique_ptr<const facebook::jsi::Buffer> UwpPreparedScriptStore::tryGetPreparedScript(
+  const facebook::jsi::ScriptSignature& scriptSignature,
+  const facebook::jsi::JSRuntimeSignature& runtimeSignature,
+  const char* prepareTag // Optional tag. For e.g. eagerly evaluated vs lazy cache.
+) noexcept
+{
+  try {
+    
+    // check if app bundle version is older than or equal to the prepared script version
+    // if true then just read the buffer from the prepared script and return
+    auto byteCodeFile = TryGetByteCodeFileSync(scriptSignature.version);
+    if (byteCodeFile == nullptr) {
+      return nullptr;
+    }
+
+    auto buffer = winrt::FileIO::ReadBufferAsync(byteCodeFile).get();
+    auto bytecodeBuffer(std::make_unique<ByteCodeBuffer>(buffer.Length()));
+    auto dataReader{ winrt::Streams::DataReader::FromBuffer(buffer) };
+    dataReader.ReadBytes(winrt::array_view<uint8_t> { &bytecodeBuffer->data()[0], &bytecodeBuffer->data()[bytecodeBuffer->size()] });
+    dataReader.Close();
+
+    return bytecodeBuffer;
+  }
+  catch (...) {
+    return nullptr;
+  }
+}
+
+void UwpPreparedScriptStore::persistPreparedScript(
+  std::shared_ptr<const facebook::jsi::Buffer> preparedScript,
+  const facebook::jsi::ScriptSignature& scriptMetadata,
+  const facebook::jsi::JSRuntimeSignature& runtimeMetadata,
+  const char* prepareTag  // Optional tag. For e.g. eagerly evaluated vs lazy cache.
+) noexcept
+{
+  persistPreparedScriptAsync(preparedScript, scriptMetadata, runtimeMetadata, prepareTag);
+}
+
+winrt::fire_and_forget UwpPreparedScriptStore::persistPreparedScriptAsync(
+  std::shared_ptr<const facebook::jsi::Buffer> preparedScript,
+  const facebook::jsi::ScriptSignature& scriptMetadata,
+  const facebook::jsi::JSRuntimeSignature& runtimeMetadata,
+  const char* prepareTag  // Optional tag. For e.g. eagerly evaluated vs lazy cache.
+)
+{
+  try {
+    co_await winrt::resume_background();
+    auto folder = winrt::ApplicationData::Current().LocalCacheFolder();
+    auto file = co_await folder.CreateFileAsync(L"app.bytecode", winrt::CreationCollisionOption::ReplaceExisting);
+    winrt::FileIO::WriteBytesAsync(file, winrt::array_view<const uint8_t>{ &preparedScript->data()[0], &preparedScript->data()[preparedScript->size()] });
+  }
+  catch (...) {
+  }
+}
+
+winrt::StorageFile UwpPreparedScriptStore::TryGetByteCodeFileSync(facebook::jsi::ScriptVersion_t version)
+{
+  try {
+    if (m_byteCodeFileAsync != nullptr) {
+      auto file = m_byteCodeFileAsync.get();
+      auto fileprops = file.GetBasicPropertiesAsync().get();
+      facebook::jsi::ScriptVersion_t byteCodeVersion = fileprops.DateModified().time_since_epoch().count();
+      if (byteCodeVersion >= version) {
```

Comment:
>byteCodeVersion >= version [](start = 10, length = 26)

Interesting .. as i understand this end up comparing the bundle timestamp to the bytecode file timestamp ? Sounds reasonable but a little creepy ..

Reply:
That's correct, it's checking timestamps. What do you find creepy? The concerns about multiple JS Bundles?